pkgs/tools/package-management/microdnf/default.nix

```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchFromGitHub, cmake, gettext, libdnf, pkgconfig, glib, libpeas, meson, libsmartcols, pcre, help2man }:
+
+stdenv.mkDerivation rec {
+  pname = "microdnf";
+  version = "3.6.0";
+
+  src = fetchFromGitHub {
+    owner = "rpm-software-management";
+    repo = pname;
+    rev = version;
+    sha256 = "0a7lc3qsnblvznzsz3544l3n84184xi85zf7c3m3jhnmpmxsg39h";
+  };
+
+  nativeBuildInputs = [ pkgconfig pcre glib libpeas cmake gettext libsmartcols help2man libdnf ];
+  buildInputs = [ meson ];
+
+  # Remove manpage generation which dies because of meson issues addressed in postInstall.
+  patches = stdenv.lib.optionals stdenv.isDarwin [ ./disable_manpage_generation.patch ];
+
+  # install_name_tool to fix a meson issue https://github.com/NixOS/nixpkgs/issues/34545
```

Comment:
The issue and those linked from there are marked as solved. Moreover, this does not sound like Meson issue if you are removing CMake code. Perhaps CMake is the build system used since you are adding it to inputs.

Reply:
Yes! I wasn't aware that it would be 'either/or'. Turns out I can remove meson from the equation.
Oddly, I still need to patch the resulting binary.

Reply:
The setup hooks of both `meson` and `cmake` override `configurePhase` if it has not been overridden before. Obviously this is mutually exclusive. But you can force one or the other using `dontUseMesonConfigure = true;` or `dontUseCmakeConfigure = true;`.

I prefer Meson for packages since it is much harder to mess up the build scripts.

Reply:
Yeah Meson seemed nicer. There was so much back and forth that I'd be happy to get this merged first and then maybe in an extra PR move it over to meson :)